checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakVisitor.java

```diff
@@ -0,0 +1,226 @@
+   * annotations on the given element.
+   *
+   * <p>Does no viewpoint-adaptation, unlike {@link
+   * CreatesMustCallForElementSupplier#getCreatesMustCallForExpressions} which does.
+   *
+   * @param elt an executable element
+   * @param mcAtf a MustCallAnnotatedTypeFactory, to source the value element
+   * @param atypeFactory a ResourceLeakAnnotatedTypeFactory
+   * @return the literal strings present in the @CreatesMustCallFor annotation(s) of that element,
+   *     substituting the default "this" for empty annotations. This method returns the empty list
+   *     iff there are no @CreatesMustCallFor annotations on elt. The returned list is always
+   *     modifiable if it is non-empty.
+   */
+  /*package-private*/ static List<String> getCreatesMustCallForValues(
+      ExecutableElement elt,
+      MustCallAnnotatedTypeFactory mcAtf,
+      ResourceLeakAnnotatedTypeFactory atypeFactory) {
+    AnnotationMirror createsMustCallForList =
+        atypeFactory.getDeclAnnotation(elt, CreatesMustCallFor.List.class);
+    List<String> result = new ArrayList<>(4);
+    if (createsMustCallForList != null) {
+      List<AnnotationMirror> createsMustCallFors =
+          AnnotationUtils.getElementValueArray(
+              createsMustCallForList,
+              mcAtf.getCreatesMustCallForListValueElement(),
+              AnnotationMirror.class);
+      for (AnnotationMirror cmcf : createsMustCallFors) {
+        result.add(getCreatesMustCallForValue(cmcf, mcAtf));
+      }
+    }
+    AnnotationMirror createsMustCallFor =
+        atypeFactory.getDeclAnnotation(elt, CreatesMustCallFor.class);
+    if (createsMustCallFor != null) {
+      result.add(getCreatesMustCallForValue(createsMustCallFor, mcAtf));
+    }
+    return result;
+  }
+
+  @Override
+  public Void visitVariable(VariableTree node, Void p) {
+    Element varElement = TreeUtils.elementFromTree(node);
+
+    if (varElement.getKind().isField()
+        && !checker.hasOption(MustCallChecker.NO_LIGHTWEIGHT_OWNERSHIP)
+        && rlTypeFactory.getDeclAnnotation(varElement, Owning.class) != null) {
+      checkOwningField(varElement);
+    }
+
+    return super.visitVariable(node, p);
+  }
+
+  /**
+   * Checks validity of a final field {@code field} with an {@code @}{@link Owning} annotation. Say
+   * the type of {@code field} is {@code @MustCall("m"}}. This method checks that the enclosing
+   * class of {@code field} has a type {@code @MustCall("m2")} for some method {@code m2}, and that
+   * {@code m2} has an annotation {@code @EnsuresCalledMethods(value = "this.field", methods =
+   * "m")}, guaranteeing that the {@code @MustCall} obligation of the field will be satisfied.
+   *
+   * @param field the declaration of the field to check
+   */
+  private void checkOwningField(Element field) {
+
+    if (checker.shouldSkipUses(field)) {
+      return;
+    }
+
+    // This value is side-effected.
+    List<String> fieldMCAnno = rlTypeFactory.getMustCallValue(field);
```

Comment:
This is a poor variable name.  It's not an annotation despite the `Anno` at the end of the variable name which usually indicates that.  It's only initially the `value` element of the `MustCall` annotation, since it is side-effected.  It is more like a set of not-yet-resolved obligations.